lzy/allocator/src/main/java/ai/lzy/allocator/services/KuberNodeService.java

```diff
@@ -0,0 +1,145 @@
+
+        if (clusterId == null || nodeName == null) {
+            String errorMessage = "Required data is not specified";
+            handleError(clientAddress, clusterId, nodeName, errorMessage);
+            return HttpResponse.status(HttpStatus.BAD_REQUEST);
+        }
+
+        LOG.info("Set node ready request; clientAddress: {}, clusterId: {}, nodeName: {}",
+            clientAddress, clusterId, nodeName);
+
+        if (readyNodes.contains(nodeName)) {
+            String errorMessage = "Node was set ready before";
+            handleError(clientAddress, clusterId, nodeName, errorMessage);
+            return HttpResponse.status(HttpStatus.BAD_REQUEST);
+        }
+
+        try {
+            if (clientAddress == null || !isNodeAddress(clientAddress, clusterId, nodeName)) {
+                String errorMessage = "Unexpected client address";
+                handleError(clientAddress, clusterId, nodeName, errorMessage);
+                return HttpResponse.status(HttpStatus.BAD_REQUEST);
+            }
+
+            if (!ensureNodeIsReady(clientAddress)) {
+                String errorMessage = "Node denied its own readiness";
+                handleError(clientAddress, clusterId, nodeName, errorMessage);
+                return HttpResponse.status(HttpStatus.BAD_REQUEST);
+            }
+
+            nodeController.addLabels(clusterId, nodeName, Map.of(NODE_READINESS_LABEL, "true"));
+        } catch (IllegalArgumentException e) {
+            handleError(clientAddress, clusterId, nodeName, e.getMessage());
+            return HttpResponse.status(HttpStatus.BAD_REQUEST);
+        } catch (Exception e) {
+            handleError(clientAddress, clusterId, nodeName, e.getMessage());
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        readyNodes.add(nodeName);
+        LOG.info("Set node ready done; nodeName: {}", nodeName);
+        return HttpResponse.ok("OK");
+    }
+
+    private boolean isNodeAddress(String address, String clusterId, String nodeName) {
+        var nodeAddresses = nodeController.getNode(clusterId, nodeName).getStatus().getAddresses().stream()
+            .map(addr -> {
+                try {
+                    return InetAddress.getByName(addr.getAddress());
+                } catch (UnknownHostException e) {
+                    LOG.warn("Failed to parse original node address {}", addr.getAddress());
+                    return null;
+                }
+            })
+            .filter(Objects::nonNull)
+            .toList();
+        try {
+            return nodeAddresses.contains(InetAddress.getByName(address));
+        } catch (UnknownHostException e) {
+            throw new IllegalArgumentException("Failed to parse client address");
+        }
+    }
+
+    private boolean ensureNodeIsReady(String host) {
+        final var requestUrl = "http://" + HostAndPort.fromParts(host, NODE_DAEMONSET_PORT) + "/isReady";
```

Comment:
let's move it to the constant pattern/template `private static final String CHECK_NODE_READINESS_TEMPLATE = "http://%s/isReady"`